lib/src/components/NxProgressBar/__tests__/NxProgressBar.test.tsx

```diff
@@ -4,191 +4,176 @@
-
-        expect(fullComponent).toHaveClassName('nx-progress-bar--full');
-        expect(fullComponent).not.toHaveClassName('nx-progress-bar--undefined');
-        expect(fullComponent.find(counterSelector)).toExist();
-        expect(fullComponent.find(labelTextSelector)).toExist();
-      }
-  );
+  const minimalProps = {
+        value: 0,
+        label: 'current progress'
+      },
+      quickRender = rtlRender<Props>(NxProgressBar, minimalProps),
+      renderEl = rtlRenderElement<Props>(NxProgressBar, minimalProps);
+
+  it('render a <label> as the top level element', function() {
+    expect(renderEl()!.tagName).toBe('LABEL');
+  });
 
-  describe('showSteps', function() {
-    it('adds an .nx-progress-bar__step-container when true', function() {
-      expect(getMountedComponent()).not.toContainMatchingElement('.nx-progress-bar__step-container');
-      expect(getMountedComponent({ showSteps: true })).toContainMatchingElement('.nx-progress-bar__step-container');
+  it('renders a <progress> element as a child', function() {
+    const el = renderEl()!,
+        progressBar = within(el).getByRole('progressbar');
+    expect(progressBar).toBeInTheDocument();
+  });
+
+  it('forwards its ref to the <progress> element', function() {
+    const ref = React.createRef<HTMLProgressElement>(),
+        el = quickRender({ ref: ref } as Partial<Props>),
+        progressBar = el.getByRole('progressbar');
+
+    expect(ref.current).toBe(progressBar);
+  });
+
+  it('passes the value prop to the progress element', function() {
+    const noProgressEl = quickRender(),
+        partialProgressEl = quickRender({ value: 50 }),
+        noProgressBar = noProgressEl.getByRole('progressbar'),
+        partialProgressBar = partialProgressEl.getByRole('progressbar');
+
+    expect(noProgressBar).toHaveAttribute('value', '0');
+    expect(partialProgressBar).toHaveAttribute('value', '50');
+  });
+
+  it('sets the max prop to 100 when unspecified and passes it to the progress element', function() {
+    const defaultEl = quickRender(),
+        customMaxEl = quickRender({ max: 200 }),
+        progressEl = defaultEl.getByRole('progressbar'),
+        customMaxProgressEl = customMaxEl.getByRole('progressbar');
+
+    expect(progressEl).toHaveAttribute('max', '100');
+    expect(customMaxProgressEl).toHaveAttribute('max', '200');
+  });
+
+  it('sets the correct counter percentage value', function() {
+    const noProgressEl = renderEl()!,
+        inProgressEl = renderEl({ value: 50 })!,
+        finishedEl = renderEl({ value: 100 })!,
+        inProgressWithMaxEl = renderEl({ value: 20, max: 40 });
+
+    expect(noProgressEl).toHaveTextContent('0%'),
+    expect(inProgressEl).toHaveTextContent('50%'),
+    expect(finishedEl).toHaveTextContent('100%');
+    expect(inProgressWithMaxEl).toHaveTextContent('50%');
+  });
+
+  it('renders a counter when showCounter is not false', function() {
+    const counterSelector = '.nx-progress-bar__counter',
```

Comment:
Rather than looking up the class name it would be better to check whether the counter text content is present

Reply:
> Rather than looking up the class name it would be better to check whether the counter text content is present

Same in the next test